notebooks/kaggle/store-sales-time-series-forecasting/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,1515 @@
+    "    trend = x.rolling(window=7, center=True, min_periods=5).mean()\n",
+    "    plt.plot(x)\n",
+    "    trend.plot(x=x, linewidth=3)\n",
+    "    plt.xlabel('date')\n",
+    "    plt.ylabel('median ' + current_num_feature)\n",
+    "    plt.title(current_num_feature)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>These trend plots prove guesses about extreme values of 'sales' and 'transactions' numerical features</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Categorical data distribution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Number of unique values:  \\n\")\n",
+    "for current_cat_feature in categorical_features:\n",
+    "    print(current_cat_feature, '-', train_data[current_cat_feature].nunique())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cat_features_hist = ['family', 'city', 'locale_name', 'store_nbr']\n",
+    "cat_features_pie = ['state', 'store_type', 'event_type', 'locale', 'transferred', 'cluster']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(25, 15))\n",
+    "for i, current_cat_column in enumerate(cat_features_pie):\n",
+    "    plt.subplot(2, 3, i + 1)\n",
+    "    n = train_data[current_cat_column].value_counts()\n",
+    "    plt.pie(n, labels=n.index, radius=1800, autopct='%1.1f%%')\n",
+    "    plt.axis('equal')\n",
+    "    plt.title(current_cat_column, color='red')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(30, 40))\n",
+    "for i, cat_feature in enumerate(cat_features_hist):\n",
+    "    plt.subplot(len(cat_features_hist), 1, i + 1)\n",
+    "    plt.xlabel(cat_feature)\n",
+    "    plt.ylabel('count')\n",
+    "    sns.countplot(x=train_data[cat_feature], alpha=0.7)\n",
+    "    plt.title(cat_feature)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Correlations between variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_copy = train_data.copy()\n",
+    "train_data_copy = pd.get_dummies(train_data_copy)\n",
+    "train_data_copy.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "corr_coefficients = (train_data_copy.corr().where(np.triu(np.ones(train_data_copy.corr().shape), k=1).astype(bool))\n",
```

Comment:
Not sure what happens here. Can you please explain?

Reply:
Here I take only an upper triangle of the correlation matrix and reshape it in this way:
![image](https://user-images.githubusercontent.com/70744513/180661020-23a3a219-f2a1-4088-859b-ccf6f51e073b.png)
After than I sort correlation coefficients and print the highest of them.